src/WebJobs.Script/Host/BlobLeaseManager.cs

```diff
@@ -123,62 +126,48 @@ private void ProcessLeaseTimerTick(object state)
 
         private async Task AcquireOrRenewLeaseAsync()
         {
-            try
+            string lockName = GetBlobName(_hostId);
+
+            DateTime requestStart = DateTime.UtcNow;
+            if (HasLease)
             {
-                DateTime requestStart = DateTime.UtcNow;
-                if (HasLease)
+                try
                 {
-                    await _lockBlob.RenewLeaseAsync(new AccessCondition { LeaseId = LeaseId });
+                    await _lockManager.RenewAsync(this.LockHandle, CancellationToken.None);
+
                     _lastRenewal = DateTime.UtcNow;
                     _lastRenewalLatency = _lastRenewal - requestStart;
                 }
-                else
+                catch
```

Comment:
No. I chose against introducing a new exception type (LockRenewalException) because I don't think it actually solves anything here. We'd still need to handle the general case. And we'd handle general and specific exception the same way. And it places an additional burden on implementors. 

---
In reply to: [123671286](https://github.com/Azure/azure-webjobs-sdk-script/pull/1615#discussion_r123671286) [](ancestors = 123671286)